.vimrc

```diff
@@ -0,0 +1,114 @@
+" Use Vim settings, rather then Vi settings. This setting must be as early as
+" possible, as it has side effects.
+set nocompatible
+
+" Leader
+let mapleader = " "
+
+" Switch between the last two files
+nnoremap <leader><leader> <c-^>
+
+" Set undo file to allow undoing even after closing and reopening vim
+silent !mkdir -p ~/.vim/undo > /dev/null 2>&1
+set undofile
+set undodir=~/.vim/undo
+
+" Softtabs, 2 spaces
+set softtabstop=2
+set tabstop=2
+set shiftwidth=2
+set shiftround
+set expandtab
+set backspace=2   " Backspace deletes like most programs in insert mode
+
+" Numbers
+set number
+set numberwidth=5
+
+" General preferences
+set nobackup
+set nowritebackup
+set noswapfile        " http://robots.thoughtbot.com/post/18739402579/global-gitignore#comment-458413287
+set history=50
+set ruler             " show the cursor position all the time
+set showcmd           " display incomplete commands
+set incsearch         " do incremental searching
+set laststatus=2      " Always display the status line
+set autowrite         " Automatically :write before running commands
+set lazyredraw        " Only redraw screen when necessary
+set diffopt+=vertical " Always use vertical diffs
+
+" Open new split panes to right and bottom
+set splitbelow
+set splitright
+
+" Quicker window movement
+nnoremap <C-j> <C-w>j
+nnoremap <C-k> <C-w>k
+nnoremap <C-h> <C-w>h
+nnoremap <C-l> <C-w>l
+
+set background=dark
```

Comment:
This may not necessarily do what we want. `background` is a way to tell vim which default color palette to load. It doesn't actually set the background color.